src/fsharp/CheckFormatStrings.fs

```diff
@@ -71,15 +72,24 @@ let parseFormatStringInternal (m:range) (g: TcGlobals) (context: FormatStringChe
 
     let specifierLocations = ResizeArray()
 
+    // For FormattableString we collect a .NET Format String with {0} etc. replacing text.  '%%' are replaced
+    // by '%', we check there are no '%' formats, and '{{' and '}}' are *not* replaced since the subsequent
+    // call to String.Format etc. will process them.
+    let dotnetFormatString = StringBuilder() 
+    let appendToDotnetFormatString (s: string) = dotnetFormatString.Append(s) |> ignore 
+    let mutable dotnetFormatStringInterpolationHoleCount = 0
+    let percentATys = ResizeArray<_>()
+
     let rec parseLoop acc (i, relLine, relCol) = 
        if i >= len then
            let argtys =
                if acc |> List.forall (fun (p, _) -> p = None) then // without positional specifiers
                    acc |> List.map snd |> List.rev
                else  
-                   failwithf "%s" <| FSComp.SR.forPositionalSpecifiersNotPermitted()
+                   raise (Failure (FSComp.SR.forPositionalSpecifiersNotPermitted()))
```

Comment:
This looks like churn - is it valuable to do it in this PR, or can it move to a different formatting PR?